pkg/networkservice/common/connect/client.go

```diff
@@ -83,14 +84,14 @@ func (u *connectClient) init() error {
 }
 
 func (u *connectClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
-	if err := u.init(); err != nil {
+	if err := u.init(clock.GetClock(ctx)); err != nil {
```

Comment:
Why not do pass ctx as previous, and use clock.GetClock(ctx) inside init method?

Reply:
Yep, we even can take clock from `u.ctx`, because it is the closest context we have and `clock.GetClock()` should return the same clock for the every given from the outside context.
Fixed.